### PR TITLE
Enable the pip package manager

### PR DESCRIPTION
**Pull request by athos-ribeiro**
- enable pkg manager in API
- use the "pip-dev-preview" flag
- enable additional configurations

### CODE REVIEW

**Comment by chmeliik** [comment]
Hijacking with a concern I missed when approving a previous PR: https://github.com/release-engineering/cachito/pull/263/files#r485635654
Not sure if the PIP_CERT env var can contain the content of the cert, it may have to be a file path (and cert provided separately)

**Comment by athos-ribeiro** [comment]
> Hijacking with a concern I missed when approving a previous PR: https://github.com/release-engineering/cachito/pull/263/files#r485635654
> Not sure if the PIP_CERT env var can contain the content of the cert, it may have to be a file path (and cert provided separately)

I replied in the closed MR. I will include the change in a different PR when adding the other configuration files to the request as well.

File: cachito/web/models.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def _validate_request_package_configs(request_kwargs, pkg_managers_names):
         )
 
     # Validate the values for each package manager configuration (e.g. packages.npm)
-    valid_package_config_keys = {"path"}
+    valid_package_config_keys = {"path", "requirements_build_files", "requirements_files"}
```
Comment (chmeliik) [inline cachito/web/models.py:419]: Should `"requirements_build_files", "requirements_files"` only be allowed for pip?

**Review by nirzari — APPROVED** [review]

File: tests/test_api_v1.py [hunk 2]
```diff
@@ -505,21 +566,9 @@ def test_create_request_invalid_dependency_replacement(
...
-            'The value of "packages.npm" must be an array of objects with the following keys: path',
-        ),
-        (
-            {"npm": [{}]},
-            ["npm"],
-            'The value of "packages.npm" must be an array of objects with the following keys: path',
-        ),
+        ({"npm": {"path": "client"}}, ["npm"], RE_INVALID_PACKAGES_VALUE),
+        ({"npm": ["path"]}, ["npm"], RE_INVALID_PACKAGES_VALUE),
+        ({"npm": [{}]}, ["npm"], RE_INVALID_PACKAGES_VALUE),
```
Comment (chmeliik) [inline tests/test_api_v1.py:571]: Could we also add a couple cases for pip here?

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

Tested, works!

**Pull request merged**